### PR TITLE
Some fixes

### PR DESCRIPTION
**Pull request by liuming50**

### CODE REVIEW

File: meta-cube/recipes-support/overc-utils/source/cube-ctl [hunk 1]
```diff
@@ -1011,9 +1011,9 @@ case "${cmd}" in
 	    out_dir="${container_dir}/"
 	fi
 
-	# if [ -n "${track_containers}" ]; then
-	#     ${SBINDIR}/overc-ctl track ${container_name} -o ${out_dir}
-	# fi
+	if [ -n "${track_containers}" ]; then
+	    ${SBINDIR}/overc-ctl track ${container_name} -o ${out_dir}
+	fi
```
Comment (zeddii) [inline meta-cube/recipes-support/overc-utils/source/cube-ctl]: This is too soon to start tracking. That's why it was moved.

For containers like docker, etc, there is a two phase addition of the container. At
the bottom, all the movement is done and then we should track it.

**Comment by zeddii** [comment]
patches 1 and 2 merged.

**Comment by liuming50** [comment]
Hi, Bruce:

Please correct me if I am wrong, but my understanding was the containers should be tracked firstly then we can do the files movements on top of the subvolumes created by 'overc-ctl track', or else all the modifications in container directories (/opt/container/cube-gw, etc.) would be removed if we do tracking later as it is after your change.

Because the current implementation of 'overc-ctl track' is as follows:
1 Check if /opt/container exists, if not, create it.
2 Check if /opt/container/${container_name} exists, if it exists, delete it. (This is mandatory since btrfs needs it to be non-existence, otherwise, btrfs would create a subvolume into that directory)
3 Check if /opt/container is being mounted on a btrfs device, if so, create a subvolume at /opt/container/${container_name}, if not, create a normal directory.

//Ming Liu

**Comment by zeddii** [comment]
That's the issue. There is a lot of data movement that happens when some container types are added, and the tracking should be something that is enabled when the final container is in place, since errors can happen at many places in the process.

Also tracking is something that will be enabled via 'cube-cfg' on an existing container, or something that can be later disabled. This is required to deal with space issues or to have steps in the process that are not tracked.

I'm seeing [error] messages thrown to the screen during the add process due to tracking having issues with btrfs and different subvol issues.

I understand the flow you are talking about above, but to allow us to work with many different types of filesystems and flows that I'm mentioning above, it needs to be something more like:

 - install a container to /opt/container/<foo>
 - optionally test to ensure it works
 - track it (i.e. create a snapshot)

So whatever is in <foo> when it is 'tracked', we take a snapshot then. Yes, there may be additional data movement, since you will probably have to move 'foo' to another location to create btrfs, or whatever other type of snapshot we are doing, and then create the FS layout and then put the container data back.

Also, it is something to consider that we store snapshots in a parallel data structure to the containers, i.e. /opt/container-snapshots/<foo>. But we may run into space issues, etc.

**Pull request closed without merging**